src/presenter/film-list-presenter.js

```diff
@@ -1,34 +1,86 @@
 import FilmCardView from '../view/film-card-view';
 import ShowMoreView from '../view/show-more-view';
 
 import FilmPopupPresenter from './film-popup-presenter';
 
+//todo - далее перенести на уровень view
 const popupRootElement = document.querySelector('body');
 const getOpenPopup = () => popupRootElement.querySelector('.film-details');
+const filmSliceCount = getFilmSlice();
 
+//todo - единый÷ класс для 3х списков, но разные параметры создания экземпляра.
 export default class FilmListPresenter {
   #siteContainers = getContainerTemplates();
   #filmSectionTemplates = getFilmSectionTemplates();
-  // todo - перенести mainListSectionComponent и mainListContainerComponent в инит - если в dataset 0 записей - меняем заголовок (view) и контейнер mainListContainerComponent не формируем
-  #mainListSectionComponent = new FilmListView(this.#filmSectionTemplates.mainList);
+  #mainListSectionComponent = new ContainerView(this.#siteContainers.filmListSection);
   #mainListContainerComponent = new ContainerView(this.#siteContainers.filmListContainer);
+  #mainListHeaderComponent = new FilmListHeaderView();
+  #showMoreComponent = new ShowMoreView();
+  #renderStartIndex = 0;
+  #renderFilmCount = filmSliceCount;
+  #filmsData = [];
+  #filmListHeaders = getFilmListHeaders();
+  #filmListContext = null;
 
   init = (contentSection, dataSet, referenceDataModel) => {
     this.contentSection = contentSection;
-    this.dataSet = dataSet;
+    this.#filmsData = dataSet;
     this.referenceDataModel = referenceDataModel;
+    // todo - пока константа. далее надо как-то получать сведения о выбранном фильтре
+    // todo - текст заголовка перенести в конструктор вьюхи
+    this.#filmListContext = this.#filmListHeaders.ALLEMPTY;
+
 
     render(this.#mainListSectionComponent, this.contentSection);
+
+    if (this.#filmsData.length === 0) {
+      this.#renderHeader(this.#mainListHeaderComponent, this.#filmListContext, this.#mainListSectionComponent.element);
+      return;
+    }
+
+    this.#renderHeader(this.#mainListHeaderComponent, this.#filmListHeaders.NOEMPTY, this.#mainListSectionComponent.element, 'visually-hidden');
+    this.#renderFilmsContainer(this.#filmsData);
+
+  };
+
+  // #renderFilmsSectionComponent = (dataSet, target) => { };
+
+  #renderHeader = (component, text, target, cssClass = null) => {
+    component.element.textContent = text;
+    component.element.classList.add(cssClass);
+    render(component, target);
+  };
+
+  #renderFilmsContainer = (dataSet) => {
     render(this.#mainListContainerComponent, this.#mainListSectionComponent.element);
 
-    // todo возможно нужен отдельный прзентер для одновременного создания карточки с попапом, а здесь вместо рендера вызывать инит
-    this.dataSet.forEach((filmItem) => this.#renderFilm(filmItem, this.referenceDataModel));
+    this.#renderFilmsSlice(this.#filmsData, this.referenceDataModel, this.#renderStartIndex, Math.min(this.#filmsData.length, filmSliceCount) );
```

Comment:
вообще-то приватным методам передавать приватные же поля не нужно - эти два друга и так живут в одном мире.
```suggestion
    this.#renderFilmsSlice(this.referenceDataModel, Math.min(this.#filmsData.length, filmSliceCount) );
```

Reply:
Не стесняйся иметь маленькие именованные методы - они служат тебе комментариями

```js
  #computeRequiredLength(filmSliceCount) => Math.min(this.#filmsData.length, filmSliceCount)
```

```suggestion
    this.#renderFilmsSlice(this.referenceDataModel, this.#computeRequiredLength(filmSliceCount) );
```

поскольку остальные параметры приватные - и их можно использовать внутри